bokeh/application/handlers/function.py

```diff
@@ -124,8 +125,12 @@ def modify_document(self, doc):
         ``False``.
 
         '''
-        self._func(doc)
-        self._safe_to_fork = False
+        try:
+            self._func(doc)
+        except Exception as e:
+            _handle_exception(self, e)
+        finally:
```

Comment:
Not sure this `finally` is really necessary in a practical sense. If there was an exception in the function, nothing is going to work anyway.

Reply:
just trying to be consistent, the code using `safe_to_fork` says "don't run if any changes have been made", but we have no way of knowing what user code did before errorring.